app/scripts/BedLikeTrack.js

```diff
@@ -336,36 +404,109 @@ class BedLikeTrack extends HorizontalTiled1DPixiTrack {
             },
             tile.tileId
           ];
+        }
 
-          // tile probably hasn't been initialized yet
-          if (!tile.texts) return;
+        if (!this.options.showTexts) continue;
+        // console.log('geneName:', geneName);
+        // tile probably hasn't been initialized yet
+        if (!tile.texts) return;
 
-          // don't draw texts for the latter entries in the tile
-          if (i >= MAX_TEXTS) continue;
+        // don't draw texts for the latter entries in the tile
+        if (i >= MAX_TEXTS) continue;
 
-          if (!tile.texts[geneName]) continue;
+        if (!tile.texts[td.uid]) continue;
 
-          const text = tile.texts[geneName];
+        const text = tile.texts[td.uid];
 
-          text.position.x = this._xScale(txMiddle);
-          // text.position.y = textYMiddle;
-          text.style = {
-            fontSize: this.textFontSize,
-            fontFamily: this.textFontFamily,
-            fill,
-          };
+        text.position.x = this._xScale(txMiddle);
+        text.position.y = rectY + rectHeight / 2;
 
-          if (!(geneInfo[3] in tile.textWidths)) {
-            text.updateTransform();
-            const textWidth = text.getBounds().width;
+        if (alreadyDrawn) {
+          text.alreadyDrawn = true;
+        }
 
-            tile.textWidths[geneInfo[3]] = textWidth;
-          }
+        text.style = Object.assign(TEXT_STYLE, { fill });
+
+        if (!(geneInfo[3] in tile.textWidths)) {
+          text.updateTransform();
+          const textWidth = text.getBounds().width;
+          const textHeight = text.getBounds().height;
+
+          tile.textWidths[geneInfo[3]] = textWidth;
+          tile.textHeights[geneInfo[3]] = textHeight;
         }
       }
     }
   }
 
+  renderTile(tile) {
+    let maxPlusRows = 1;
+    let maxMinusRows = 1;
+
+    for (const tile1 of this.visibleAndFetchedTiles()) {
```

Comment:
What's the difference between `tile` and `tile1`? Is `renderTile(tile)` just rendering `tile` or is it rendering all visible and fetched tiles? I am slightly confused.

Reply:
Oh is `tile1` just an already rendered tile? I guess I know what's going on. It might still be nice to rename `tile1` into something more meaningful.

Reply:
Yeah, it's just used to iterate over all the other tiles so that we can find out what the maximum number of rows is.

Reply:
I'll rename it.